tests/test/transport/test_message_endpoint_client.cpp

```diff
@@ -224,6 +225,83 @@ TEST_CASE_METHOD(SchedulerTestFixture,
     }
 }
 
-#endif
+TEST_CASE_METHOD(SchedulerTestFixture, "Test direct messaging", "[transport]")
+{
+    std::string expected = "Direct hello";
+    const uint8_t* msg = BYTES_CONST(expected.c_str());
+
+    std::string inprocLabel = "direct-test";
+
+    AsyncDirectSendEndpoint sender(inprocLabel, TEST_PORT);
+    sender.send(msg, expected.size());
+
+    AsyncDirectRecvEndpoint receiver(inprocLabel);
+
+    std::string actual;
+    SECTION("Recv with size")
+    {
+        faabric::transport::Message recvMsg =
+          receiver.recv(expected.size()).value();
+        actual = std::string(recvMsg.data(), recvMsg.size());
+    }
+
+    SECTION("Recv no size")
+    {
+        faabric::transport::Message recvMsg = receiver.recv().value();
+        actual = std::string(recvMsg.data(), recvMsg.size());
+    }
+
+    REQUIRE(actual == expected);
+}
+
+TEST_CASE_METHOD(SchedulerTestFixture,
```

Comment:
For my particular use case, I'd be interested in seeing a stress test with various threads producing and various threads consuming to different in proc labels (each producer matched to one consumer).

Feel free to ignore, but I think it could be a relevant stress test.

Reply:
Yes good idea, have updated.